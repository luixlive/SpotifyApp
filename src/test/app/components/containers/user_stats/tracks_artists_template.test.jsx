import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import initialState from './../../../../test_utils/initial_state';
import TracksArtistsTemplate, {
  PureTracksArtistsTemplate,
/* eslint-disable-next-line max-len */
} from './../../../../../app/components/containers/user_stats/tracks_artists_template';
import {
  tracksArtistsTemplate as componentProps,
} from './../../../../test_utils/components_props';

describe('App Components - TracksArtistsTemplate', () => {
  const mockStore = configureStore();
  let store;
  beforeAll(() => {
    store = mockStore(initialState);
  });

  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders device no mobile type tracks', () => {
      const rendered = renderer.create((
        <Provider store={store}>
          <PureTracksArtistsTemplate {...props} />
        </Provider>
      )).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders device mobile type tracks', () => {
      props.deviceMobile = true;
      const rendered = renderer.create((
        <Provider store={store}>
          <PureTracksArtistsTemplate {...props} />
        </Provider>
      )).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders device no mobile type artists', () => {
      props.type = 'artists';
      const rendered = renderer.create((
        <Provider store={store}>
          <PureTracksArtistsTemplate {...props} />
        </Provider>
      )).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders device mobile type artists', () => {
      props.deviceMobile = true;
      props.type = 'artists';
      const rendered = renderer.create((
        <Provider store={store}>
          <PureTracksArtistsTemplate {...props} />
        </Provider>
      )).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('returns null when renderPage and there is an invalid type', () => {
      const props = _.cloneDeep(componentProps);
      props.type = 'unknown';
      const wrapper = shallow(<PureTracksArtistsTemplate {...props} />);
      expect(wrapper.instance().renderPage()).toBeNull();
    });
  });

  describe('Provider', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = mount((
        <Provider store={store}>
          <TracksArtistsTemplate type="tracks" />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureTracksArtistsTemplate).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureTracksArtistsTemplate).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
