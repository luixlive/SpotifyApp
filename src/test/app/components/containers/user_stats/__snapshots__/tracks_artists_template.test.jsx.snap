// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - TracksArtistsTemplate Snapshots renders device mobile type artists 1`] = `
<div>
  <div>
    <div
      className="ui green segment"
    >
      <form
        action={undefined}
        className="ui form"
        onSubmit={[Function]}
      >
        <h3
          className="ui header"
        >
          Since
        </h3>
        <div
          className="grouped fields"
        >
          <div
            className="field"
          >
            <div
              className="ui checked radio checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <input
                checked={true}
                className="hidden"
                id={undefined}
                name="timeRange"
                onClick={[Function]}
                readOnly={true}
                tabIndex={0}
                type="radio"
                value={undefined}
              />
              <label
                htmlFor={undefined}
              >
                last month
              </label>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="ui radio checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                id={undefined}
                name="timeRange"
                onClick={[Function]}
                readOnly={true}
                tabIndex={0}
                type="radio"
                value={undefined}
              />
              <label
                htmlFor={undefined}
              >
                last six months
              </label>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="ui radio checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                id={undefined}
                name="timeRange"
                onClick={[Function]}
                readOnly={true}
                tabIndex={0}
                type="radio"
                value={undefined}
              />
              <label
                htmlFor={undefined}
              >
                beginning of times
              </label>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="ui basic segment"
    >
      <div
        className="ui grid"
      />
    </div>
  </div>
</div>
`;

exports[`App Components - TracksArtistsTemplate Snapshots renders device mobile type tracks 1`] = `
<div>
  <div>
    <div
      className="ui green segment"
    >
      <form
        action={undefined}
        className="ui form"
        onSubmit={[Function]}
      >
        <h3
          className="ui header"
        >
          Since
        </h3>
        <div
          className="grouped fields"
        >
          <div
            className="field"
          >
            <div
              className="ui checked radio checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <input
                checked={true}
                className="hidden"
                id={undefined}
                name="timeRange"
                onClick={[Function]}
                readOnly={true}
                tabIndex={0}
                type="radio"
                value={undefined}
              />
              <label
                htmlFor={undefined}
              >
                last month
              </label>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="ui radio checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                id={undefined}
                name="timeRange"
                onClick={[Function]}
                readOnly={true}
                tabIndex={0}
                type="radio"
                value={undefined}
              />
              <label
                htmlFor={undefined}
              >
                last six months
              </label>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="ui radio checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                id={undefined}
                name="timeRange"
                onClick={[Function]}
                readOnly={true}
                tabIndex={0}
                type="radio"
                value={undefined}
              />
              <label
                htmlFor={undefined}
              >
                beginning of times
              </label>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="ui basic segment"
    >
      <div
        className="ui grid"
      />
    </div>
  </div>
</div>
`;

exports[`App Components - TracksArtistsTemplate Snapshots renders device no mobile type artists 1`] = `
<div>
  <div
    className="ui two column grid"
  >
    <div
      className="four wide column"
    >
      <div
        className="ui green segment"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <h3
            className="ui header"
          >
            Since
          </h3>
          <div
            className="grouped fields"
          >
            <div
              className="field"
            >
              <div
                className="ui checked radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  checked={true}
                  className="hidden"
                  id={undefined}
                  name="timeRange"
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                  value={undefined}
                />
                <label
                  htmlFor={undefined}
                >
                  last month
                </label>
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  id={undefined}
                  name="timeRange"
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                  value={undefined}
                />
                <label
                  htmlFor={undefined}
                >
                  last six months
                </label>
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  id={undefined}
                  name="timeRange"
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                  value={undefined}
                />
                <label
                  htmlFor={undefined}
                >
                  beginning of times
                </label>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="twelve wide column"
    >
      <div
        className="ui basic segment"
      >
        <div
          className="ui grid"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`App Components - TracksArtistsTemplate Snapshots renders device no mobile type tracks 1`] = `
<div>
  <div
    className="ui two column grid"
  >
    <div
      className="four wide column"
    >
      <div
        className="ui green segment"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <h3
            className="ui header"
          >
            Since
          </h3>
          <div
            className="grouped fields"
          >
            <div
              className="field"
            >
              <div
                className="ui checked radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  checked={true}
                  className="hidden"
                  id={undefined}
                  name="timeRange"
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                  value={undefined}
                />
                <label
                  htmlFor={undefined}
                >
                  last month
                </label>
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  id={undefined}
                  name="timeRange"
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                  value={undefined}
                />
                <label
                  htmlFor={undefined}
                >
                  last six months
                </label>
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  id={undefined}
                  name="timeRange"
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                  value={undefined}
                />
                <label
                  htmlFor={undefined}
                >
                  beginning of times
                </label>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="twelve wide column"
    >
      <div
        className="ui basic segment"
      >
        <div
          className="ui grid"
        />
      </div>
    </div>
  </div>
</div>
`;
